src/lib.rs

```diff
@@ -332,6 +333,51 @@ const RUNNING_STATS_EVERY: usize = 15;
 /// Constant defining Goose's default port when running a Gaggle.
 const DEFAULT_PORT: &str = "5115";
 
+static APP_USER_AGENT: &str = concat!(env!("CARGO_PKG_NAME"), "/", env!("CARGO_PKG_VERSION"));
+
+// Share CLIENT object globally.
```

Comment:
@LionsAd This is the largest change in this PR -- moving mutable elements from `GooseClient` into a new global mutable per-thread `GooseClientState`